### PR TITLE
authentication with devise and devise_token_auth

### PR DESCRIPTION
**Pull request by mgonc**
implements authentication token based with devise and some end points
that will work as api to citron and thaithi applications.
register, sign_up and password recovery implemented.

### CODE REVIEW

**Comment by mgonc** [comment]
This version of application breaks olds versions of thorn and citron. I recommend a clean setup, generating a new database for users.

**Review by michelboaventura — APPROVED** [review]

File: config/deploy.rb [hunk 1]
```diff
@@ -5,14 +5,14 @@
...
-set :repo_url, 'git@github.com:eubr-bigsea/torn.git'
+set :application, 'thorn'
+set :repo_url, 'git@github.com:eubr-bigsea/thorn.git'
 
 # Default branch is :master
 # ask :branch, `git rev-parse --abbrev-ref HEAD`.chomp
 
 # Default deploy_to directory is /var/www/my_app_name
-set :deploy_to, '/home/ec2-user/torn'
+set :deploy_to, '/home/ec2-user/thorn'
```
Comment (guimaluf) [inline config/deploy.rb:15]: this could be provided via environment variable instead of hardcoded

File: config/deploy/production.rb [hunk 2]
```diff
@@ -6,7 +6,7 @@
 # server 'example.com', user: 'deploy', roles: %w{app db web}, my_property: :my_value
 # server 'example.com', user: 'deploy', roles: %w{app web}, other_property: :other_value
 # server 'db.example.com', user: 'deploy', roles: %w{db}
-server 'torn', user: 'ec2-user', roles: %w{app db web}
+server 'thorn', user: 'ec2-user', roles: %w{app db web}
```
Comment (guimaluf) [inline config/deploy/production.rb:9]: same here

File: config/unicorn/production.rb [hunk 3]
```diff
@@ -1,8 +1,8 @@
-working_directory "/home/ec2-user/torn/current"
+working_directory "/home/ec2-user/thorn/current"
 worker_processes 4
-listen "/tmp/torn.sock", backlog: 64
+listen "/tmp/thorn.sock", backlog: 64
 timeout 60
-pid "/home/ec2-user/torn/shared/tmp/pids/unicorn.pid"
+pid "/home/ec2-user/thorn/shared/tmp/pids/unicorn.pid"
```
Comment (guimaluf) [inline config/unicorn/production.rb:5]: and here

**Review by guimaluf — APPROVED** [review]

**Pull request merged**